dlp/deid.py

```diff
@@ -0,0 +1,548 @@
+    table_item = {
+        'table': {
+            'headers': csv_headers,
+            'rows': csv_rows
+        }
+    }
+
+    # Construct date shift config
+    date_shift_config = {
+        'lower_bound_days': lower_bound_days,
+        'upper_bound_days': upper_bound_days
+    }
+
+    # If using Cloud KMS key
+    if context_field_id and key_name and wrapped_key:
+        import base64
+        date_shift_config['context'] = {'name': context_field_id}
+        date_shift_config['crypto_key'] = {
+            'kms_wrapped': {
+                'wrapped_key': base64.b64decode(wrapped_key),
+                'crypto_key_name': key_name
+            }
+        }
+    elif context_field_id or key_name or wrapped_key:
+        raise StandardError("""You must set either ALL or NONE of
+        [context_field_id, key_name, wrapped_key]!""")
+
+    # Construct Deidentify Config
+    deidentify_config = {
+        'record_transformations': {
+            'field_transformations': [
+                {
+                    'fields': date_fields,
+                    'primitive_transformation': {
+                        'date_shift_config': date_shift_config
+                    }
+                }
+            ]
+        }
+    }
+
+    # Write to CSV helper methods
+    def write_header(header):
+        return header.name
+
+    def write_data(data):
+        return data.string_value or '%s/%s/%s' % (data.date_value.month,
+                                                  data.date_value.day,
+                                                  data.date_value.year)
+
+    # Call the API
+    response = dlp.deidentify_content(parent,
+                                      deidentify_config=deidentify_config,
+                                      item=table_item)
+    # Write results to CSV file
+    with open(output_csv_file, 'wb') as csvfile:
+        write_file = csv.writer(csvfile, delimiter=',')
+        write_file.writerow(map(write_header, response.item.table.headers))
+        for row in response.item.table.rows:
+            write_file.writerow(map(write_data, row.values))
+    # Print status
+    print('Successfully saved date-shift output to {}'.format(
+                output_csv_file))
+# [END deidentify_with_date_shift]
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description=__doc__)
+    subparsers = parser.add_subparsers(
+        dest='content', help='Select how to submit content to the API.')
+
+    mask_parser = subparsers.add_parser(
+        'deid_mask',
+        help='Deidentify sensitive data in a string by masking it with a '
+             'character.')
+    mask_parser.add_argument('item', help='The string to deidentify.')
+    mask_parser.add_argument(
+        '-n', '--number_to_mask', type=int,
+        help='The maximum number of sensitive characters to mask in a match. '
+        'If omitted the request or set to 0, the API will mask any mathcing '
+        'characters.')
```

Comment:
add `default=0` here; otherwise `None` will override the function's default. In this case it scarcely matters because None and 0 are treated similarly, but we might as well be consistent.

Reply:
done